### ISSUE TITLE
MT: wakeup scheduler sleeping on event loop (blocking)

### ISSUE THREAD

**Issue by ysbaddaden** [ysbaddaden/execution_context#2]
A scheduler will park its associated thread (mutex + condition variable) when its event loop is empty (there still is one EL per thread for backward compatibility reasons with IO::Evented), but when there is a long pending event in the event loop then the scheduler will block on it (in its `#run_loop`) and never park the thread.

The scheduler hence puts itself into an "idle mode" until the event (or another event) is triggered, blocking the thread from doing anything during that time.

An extreme case I just had with MT:2 is scheduler B stealing the signal loop from scheduler A during startup, the fiber immediately waits on the signal fd to be readable. Not having any other fiber during startup, the scheduler will block on the event loop until a signal is received, which can be forever.

**TODO**: there should either be a single idling mechanism, or there should be a check for the "idle state", and in both cases a mean to interrupt the EL.

**Comment by ysbaddaden** [comment]
The bug is fixed but it still incurs some performance impact (lock with push/pop/delete over a Deque). A Crystal::PointerLinkedList might fare a little better (especially delete), but ideally we shouldn't have to check the blocked/parked on each and every enqueue.

**Comment by ysbaddaden** [comment]
closed by #6
